federation/README.md

```diff
@@ -0,0 +1,502 @@
+
+On the other side, the stock market service provider's SPIRE Server is configured in a similar fashion:
+```hcl
+server {
+    .
+    .
+    trust_domain = "stockmarket.example"
+    .
+    .
+
+    federation {
+        bundle_endpoint {
+            address = "0.0.0.0"
+            port = 8443
+        }
+    }
+}
+```
+
+At this point, both SPIRE Servers expose their federation endpoints to provide their trust bundles, but none of them knows how to reach each other's federation endpoint. 
+
+## Configure a Federation Endpoint Using Web PKI Authentication
+
+We are going to assume that only the broker's SPIRE Server will use Web PKI authentication for its federation endpoint. The stock market SPIRE Server will still use SPIFFE authentication. Hence, the stock market SPIRE Server configuration remains the same as seen in the previous section.
+
+Then, to configure the broker's SPIRE Server bundle endpoint, we configure the `federation` section as follows:
+
+```hcl
+server {
+    .
+    .
+    trust_domain = "broker.example"
+    .
+    .
+
+    federation {
+        bundle_endpoint {
+            address = "0.0.0.0"
+            port = 443
+            acme {
+                domain_name = "broker.example"
+                email = "some@email.com"
+                tos_accepted = true
+            }   
+        }
+    }
+}
+```
+This will publish the federation endpoint in any IP address at port 443. We use port 443 because we will be using Let's Encrypt as our ACME provider (this is used by default, if you want to use a different one then you must set the `directory_url` configurable). Note that `tos_accepted` was set to `true`, meaning that we accept the terms of service of our ACME provider, which in turn is needed when using Let's Encrypt.
+
+For SPIFFE Federation using Web PKI to work, you must own the domain specified for `domain_name` (`broker.example` in our example) and the domain must resolve to the SPIRE Server exposing the federation bundle endpoint.
+
+# Configure SPIRE Servers to Retrieve Trust Bundles From Each Other
+
+After configuring federation endpoints, the next step to enable SPIFFE federation is to configure the SPIRE Servers to find the trust bundles for other trust domains. The `federates_with` configuration option in `server.conf` is where you specify the endpoint of the other trust domain. The configuration of this section has some slight differences when using the different methods of authentication.
+
+## Configure Trust Bundle Location Using SPIFFE Authentication
+
+As we saw previously, the SPIRE Server of the stock market service provider has its federation endpoint listening on port `8443` at any IP address. We will also assume that `spire-server-stock` is a DNS name that resolves to the stock market service's SPIRE Server IP address. Then, the broker's SPIRE Server must be configured with the following `federate_with` section:
```

Comment:
What I meant is that for the docker compose example we don't have to use a FQDN there, instead we just need the hostname because all the containers are running on the same network. But, in a more realistic scenario it would likely be a FQDN.

Reply:
Ah, so you **do** mean "DNS name". It might be good to clarify this for users. See if this sentence makes sense. Feel free to suggest changes.
```suggestion
As we saw previously, the SPIRE Server of the stock market service provider has its federation endpoint listening on port `8443` at any IP address. We will also assume that `spire-server-stock` is a DNS name that resolves to the stock market service's SPIRE Server IP address. (The Docker Compose demo here uses the hostname `spire-server-stock`, but in typical usage you would specify a FQDN.) Then, the broker's SPIRE Server must be configured with the following `federate_with` section:
```